bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/channel/ChannelTypeResource.java

```diff
@@ -137,8 +129,8 @@ private ChannelTypeDTO convertToChannelTypeDTO(ChannelType channelType, Locale l
                 parameterGroups, channelType.getState(), channelType.getTags(), channelType.isAdvanced());
     }
 
-    private Set<ChannelTypeDTO> convertToChannelTypeDTOs(List<ChannelType> channelTypes, Locale locale) {
-        Set<ChannelTypeDTO> channelTypeDTOs = new HashSet<>();
+    private List<ChannelTypeDTO> convertToChannelTypeDTOs(List<ChannelType> channelTypes, Locale locale) {
+        List<ChannelTypeDTO> channelTypeDTOs = new ArrayList<>();
```

Comment:
You are right in saying that it's now not guaranteed in the REST resource anymore. Nevertheless, it is using what it gets from `channelTypeRegistry.getChannelTypes(locale)`. I'm not sure right now whether it is explicitly handled there as of today, but that would be the right place to do so.